### PR TITLE
Angular 17 support

### PR DESCRIPTION
**Pull request by pavlikxor**
Angular v.17 support added

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,32 +35,32 @@
...
   "dependencies": {
-    "@angular/animations": "^13.3.5",
-    "@angular/common": "^13.1.2",
-    "@angular/compiler": "^13.3.5",
-    "@angular/core": "^13.1.2",
-    "@angular/forms": "^13.3.5",
-    "@angular/platform-browser": "^13.3.5",
-    "@angular/platform-browser-dynamic": "^13.3.5",
-    "@angular/router": "^13.3.5",
+    "@angular/animations": "^17.0.4",
```
Comment (neodescis) [inline package.json:38]: Are we sure we want to do this? The library will *not* be backward compatible if we are building with Angular 17!

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ cache:
 addons:
   chrome: stable
 before_script:
-  - npm install -g @angular/cli@9.0.4
+  - npm install -g @angular/cli@latest
```
Comment (neodescis) [inline .travis.yml:10]: This works until Angular introduces breaking changes to the cli package...

File: projects/picker/src/lib/dialog/dialog-ref.class.ts [hunk 3]
```diff
@@ -29,20 +29,21 @@ export class OwlDialogRef<T> {
     public componentInstance: T;
 
     /** Whether the user is allowed to close the dialog. */
-    public disableClose = this.container.config.disableClose;
+    public disableClose = true;
```
Comment (neodescis) [inline projects/picker/src/lib/dialog/dialog-ref.class.ts:32]: Why this change?

File: projects/picker/package.json [hunk 4]
```diff
@@ -24,9 +24,9 @@
     "access": "public"
   },
   "peerDependencies": {
-    "@angular/common": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0",
-    "@angular/core": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0",
-    "@angular/cdk": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0"
+    "@angular/common": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0 || ^17.0.0",
```
Comment (neodescis) [inline projects/picker/package.json:27]: IMO, we should just say `>=13.0.0`

File: .github/workflows/ci_test.yml [hunk 5]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v1
         with:
-            node-version: '14.16.1'
+            node-version: '16.16.0'
```
Comment (neodescis) [inline .github/workflows/ci_test.yml:29]: I thought Angular 17 required node 18

File: projects/picker/package.json [hunk 6]
```diff
@@ -24,9 +24,9 @@
     "access": "public"
   },
   "peerDependencies": {
-    "@angular/common": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0",
-    "@angular/core": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0",
-    "@angular/cdk": "^13.0.3 || ^14.0.0 || ^15.0.0 || ^16.0.0"
+    "@angular/common": "^17.0.0",
```
Comment (neodescis) [inline projects/picker/package.json:27]: Do we want to advertise as `>=17.0.0`? Libraries are generally forward-compatible, and changing this to `>=` would make things easier for consumers of the library going forward.

File: projects/picker/src/lib/dialog/dialog-ref.class.ts (hunk 3 above)
Comment (neodescis) [reply projects/picker/src/lib/dialog/dialog-ref.class.ts:33]: Ah, I guess I missed that this is now set to `this.container.config.disableClose` in the constructor instead. Looks good!

File: projects/picker/package.json (hunk 6 above)
Comment (neodescis) [reply projects/picker/package.json:27]: So, I meant literally `>=17.0.0` instead of `^17.0.0`

This would allow the library to be used with Angular 18 when it's released, with no changes to the library.

File: projects/picker/package.json [hunk 7]
```diff
@@ -24,9 +24,9 @@
...
+    "@angular/core": "^17.0.0",
```
Comment (neodescis) [inline projects/picker/package.json:28]:
```suggestion
    "@angular/core": ">=17.0.0",
```

File: projects/picker/package.json [hunk 8]
```diff
@@ -24,9 +24,9 @@
...
+    "@angular/cdk": "^17.0.0"
```
Comment (neodescis) [inline projects/picker/package.json:29]:
```suggestion
    "@angular/cdk": ">=17.0.0"
```

**Comment by danielmoncada** [comment]
fixes https://github.com/danielmoncada/date-time-picker/issues/188
